gm4_sunken_treasure/data/sunken_treasure/functions/spawn_treasure.mcfunction

```diff
@@ -7,41 +7,41 @@ scoreboard players set 200 gm4_st_random 200
+execute if score treasureItem gm4_st_random matches 50..79 run data merge entity @s {Item:{id:"minecraft:seagrass"}}
+execute if score treasureItem gm4_st_random matches 80..104 run summon squid ~ ~ ~
+execute if score treasureItem gm4_st_random matches 105..120 run data merge entity @s {Item:{id:"minecraft:mossy_cobblestone"}}
+execute if score treasureItem gm4_st_random matches 121..135 run data merge entity @s {Item:{id:"minecraft:kelp"}}
+execute if score treasureItem gm4_st_random matches 136..149 run data merge entity @s {Item:{id:"minecraft:sea_pickle"}}
+execute if score treasureItem gm4_st_random matches 150..159 run data merge entity @s {Item:{id:"minecraft:prismarine_crystals"}}
+execute if score treasureItem gm4_st_random matches 160..169 run data merge entity @s {Item:{id:"minecraft:prismarine_shard"}}
+execute if score treasureItem gm4_st_random matches 170..175 run data merge entity @s {Item:{id:"minecraft:scute"}}
 execute if score treasureItem gm4_st_random matches 176..180 run data merge entity @s {Item:{id:"minecraft:experience_bottle"}}
 execute if score treasureItem gm4_st_random matches 181..185 run data merge entity @s {Item:{id:"minecraft:gold_ingot"}}
 execute if score treasureItem gm4_st_random matches 186..190 run summon guardian ~ ~ ~
-execute if score treasureItem gm4_st_random matches 191..195 run data merge entity @s {Item:{id:"minecraft:leather_boots"}}
-execute if score scuba_gear gm4_modules matches 1.. if score treasureItem gm4_st_random matches 195 run data merge entity @s {Item:{id:"golden_chestplate",Count:1b,Damage:110s,tag:{gm4_scuba_gear:{item:"tank"},display:{Name:"{\"text\":\"SCUBA Tank\",\"italic\":\"false\"}"},AttributeModifiers:[{AttributeName:"generic.movementSpeed",Name:"generic.movementSpeed",Amount:-.01,Operation:0,UUIDLeast:189444,UUIDMost:911382,Slot:"chest"},{AttributeName:"generic.armor",Name:"generic.armor",Amount:4,Operation:0,UUIDLeast:778609,UUIDMost:847530,Slot:"chest"}]}}}
-execute if score treasureItem gm4_st_random matches 196..198 run data merge entity @s {Item:{id:"minecraft:wet_sponge"}}
+execute if score treasureItem gm4_st_random matches 191..193 run data merge entity @s {Item:{id:"minecraft:golden_boots",tag:{Damage:62s,display:{Name:"{\"text\":\"Waders\",\"italic\":false}"},Enchantments:[{id:"minecraft:depth_strider",lvl:1}]}}}
+execute if score scuba_gear gm4_modules matches 1.. if score treasureItem gm4_st_random matches 193 run data merge entity @s {Item:{id:"leather_boots",tag:{Damage:61s,gm4_scuba_gear:{item:"flippers"},display:{Name:"{\"text\":\"Flippers\",\"italic\":\"false\"}",color:11009871},AttributeModifiers:[{AttributeName:"generic.armor",Name:"generic.armor",Amount:1,Operation:0,UUIDLeast:167143,UUIDMost:881952,Slot:"feet"},{AttributeName:"generic.movementSpeed",Name:"generic.movementSpeed",Amount:-.04,Operation:0,UUIDLeast:942428,UUIDMost:579389,Slot:"feet"}]}}}

```

Comment:
Why are you checking to see if the module is installed here and on line 46? If this file is running, scuba_gear should always be 1.. in gm4_modules